### PR TITLE
translation: Update the figures in the chapter of preface and introduction

### PR DESCRIPTION
**Pull request by krahets**
If this pull request (PR) pertains to **Chinese-to-English translation**, please confirm that you have read the contribution guidelines and complete the checklist below:

- [x] This PR represents the translation of a single, complete document, or contains only bug fixes.
- [x] The translation accurately conveys the original meaning and intent of the Chinese version. If deviations exist, I have provided explanatory comments to clarify the reasons.

### CODE REVIEW

**Review by K3v123 — APPROVED** [review]
looks good to me

File: docs-en/chapter_introduction/algorithms_are_everywhere.assets/binary_search_dictionary_step2.png
Comment (yuelinxin) [inline docs-en/chapter_introduction/algorithms_are_everywhere.assets/binary_search_dictionary_step2.png:1]: should the numbering and "the first half is excluded" be bolded? as does in the Chinese version?

File: docs-en/chapter_introduction/algorithms_are_everywhere.assets/binary_search_dictionary_step3.png
Comment (yuelinxin) [inline docs-en/chapter_introduction/algorithms_are_everywhere.assets/binary_search_dictionary_step3.png:1]: same here

File: docs-en/chapter_introduction/algorithms_are_everywhere.assets/binary_search_dictionary_step4.png
Comment (yuelinxin) [inline docs-en/chapter_introduction/algorithms_are_everywhere.assets/binary_search_dictionary_step4.png:1]: same here

**Review by yuelinxin — COMMENTED** [review]
Great work translating the figures, this is not easy work! See if the bolded text is necessary as commented below.

File: docs-en/chapter_introduction/algorithms_are_everywhere.assets/binary_search_dictionary_step2.png
Comment (krahets) [reply docs-en/chapter_introduction/algorithms_are_everywhere.assets/binary_search_dictionary_step2.png:1]: Yes, I'll fix it

**Comment by krahets** [comment]
@yuelinxin @K3v123 I also modified "Dictionaries sorted alphabetically" to "Dictionary sorted alphabetically". Please check the correctness

**Review by K3v123 — APPROVED** [review]

**Comment by krahets** [comment]
This PR will be merged without the approval of @yuelinxin. All the previous comments were addressed. Please let me know if there are any more mistakes, thx!

**Review by yuelinxin — COMMENTED** [review]
great work, thank you!

File: docs-en/chapter_introduction/algorithms_are_everywhere.assets/binary_search_dictionary_step2.png
[the remark this answers is not in the document]
Comment (krahets) [reply docs-en/chapter_introduction/algorithms_are_everywhere.assets/binary_search_dictionary_step2.png:1]: I'll fix it. Thx!

**Pull request merged**